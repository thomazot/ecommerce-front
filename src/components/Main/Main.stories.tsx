import { Story, Meta } from '@storybook/react'
import Main from '.'

export default {
  title: 'Components/Main',
  component: Main,
  args: {
    children: 'Component Main'
  }
} as Meta

export const Default: Story = (args) => <Main {...args} />
