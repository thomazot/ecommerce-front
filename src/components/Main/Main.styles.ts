import styled from 'styled-components'
import tw from 'twin.macro'

export const Wrapper = styled.main`
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`

export const Logo = styled.img`
  width: 25rem;
  margin-bottom: 2rem;
`

export const Title = styled.h1`
  ${tw`
    text-3xl 
    font-bold 
    tracking-tight 
    text-gray-900
  `}
`

export const Description = styled.h2`
  font-size: 2rem;
  font-weight: 400;
`

export const Illustration = styled.img`
  margin-top: 3rem;
  width: min(30rem, 100%);
`
