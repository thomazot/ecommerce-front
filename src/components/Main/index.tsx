const Main = () => (
  <main>
    <h1>react avanÃ§ados</h1>
  </main>
)

export default Main
