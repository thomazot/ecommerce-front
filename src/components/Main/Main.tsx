import * as S from './Main.styles'

export type MainProps = {
  children?: React.ReactNode | React.ReactNode[]
}

const Main: React.FC<MainProps> = ({ children }) => (
  <S.Container>
    <div className="mx-auto max-w-7xl px-4 sm:px-6 lg:px-8">{children}</div>
  </S.Container>
)

export default Main
