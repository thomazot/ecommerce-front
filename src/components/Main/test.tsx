import { screen, render } from '@testing-library/react'

import Main from './'

describe('<Main />', () => {
  it('should render the heading', () => {
    render(<Main />)

    expect(
      screen.getByRole('heading', { name: /react avan√ßado/i })
    ).toBeInTheDocument()
  })
})
