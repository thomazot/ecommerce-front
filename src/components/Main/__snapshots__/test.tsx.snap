// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
<main>
  <h1>
    react avan√ßados
  </h1>
</main>
`
